### DISCUSSION TITLE
Is there anything I can do to make GIA more responsive on my Mac mini (Late 2012)

### DISCUSSION THREAD

**Discussion by ourstanley** [Ascoware/get-iplayer-automator#357] [Q&A]
So I'm running GIA on a Mac Mini (Late 2012) 2.5Ghz Dual Core Intel Core i5 with 16GB RAM running MacOS Catalina, and I find that the application is constantly beach balling. I cannot expand the RAM any more, and I am running the computer off of an external SSD (which improved performance greatly). Is there anything else I could be doing that would improve the performance of the program on this computer?

**Comment by skovatch** [comment]
I don't think it's a function of memory or processor but rather bad code. Beachballing is usually a sign that the main thread is blocked, preventing it from handling events. I have seen this too, but haven't had the spare cycles to investigate it.

**Reply by skovatch** [reply]
Oh, I see the problem... there is some code that launches a task and then sits and waits for it to finish. This blocks the main thread, and quite frankly, shouldn't be needed.

**Reply by ourstanley** [reply]
OK, thanks - because it's only once in a while that I browse the iPlayer site to grab a few shows to download, so it seems like I end up with 30-40 shows to download at a time and it takes me the best part of a week to download them because of all the wasted cycles. If you are able to clean that up in one of the next few releases that would be much appreciated.
